relays/client-substrate/src/client.rs

```diff
@@ -52,6 +53,15 @@ pub struct Client<C: Chain> {
 	client: RpcClient,
 	/// Genesis block hash.
 	genesis_hash: C::Hash,
+	/// If several tasks are submitting their transactions simultaneously using `sumbit_signed_extrinsic`
+	/// method, they may get the same transaction nonce. So one of transactions will be rejected
+	/// from the pool. This lock is here to prevent situations like that.
+	///
+	/// Note that it's just lock, not the `HashMap<AccountId, Index>`. This is because querying index
+	/// does not necessarily mean that transaction with this nonce will be included into the block. If
+	/// it'll be rejected or dropped, we may end up with incorrect (future) nonce in the map => infinite
+	/// relay loop.
```

Comment:
This is an implementation detail, so I don't think this should be part of the public doc comments. We should either turn it into "regular" comments, or just remove it